### PR TITLE
[BartTokenizer] add prepare s2s batch

### PR DESCRIPTION
**Pull request by patil-suraj**
This PR adds prepare_seq2seq_batch method to BartTokenizer as per the proposal in #6080

@sshleifer

### CODE REVIEW

**Review by sshleifer — APPROVED** [review]
LGTM, nice tests!

File: src/transformers/tokenization_bart.py
Comment (sgugger) [inline src/transformers/tokenization_bart.py:58]:
```suggestion
        """
        Prepare a batch that can be passed directly to an instance of BartModel.
```
(nit)

**Review by sgugger — APPROVED** [review]
Thanks for the PR!

Lots of nits on the docs: in general if the argument you are documenting is passed along to another method, don't hesitate to copy-paste the docstring from that method. And when documenting an argument, don't use abbreviations and make full sentences :-)

**Comment by patil-suraj** [comment]
Thanks @sgugger  for theses helpful suggestions!. Will keep these in mind for future PRs.

**Comment by patil-suraj** [comment]
@sgugger , can you help me with the build_doc failure ? Thanks!

**Comment by sgugger** [comment]
Fixed, you needed to have the beginning of the docstrings on a new line for sphinx to understand the indentation.

**Comment by patil-suraj** [comment]
> Fixed, you needed to have the beginning of the docstrings on a new line for sphinx to understand the indentation.

Thanks @sgugger !

**Review by LysandreJik — COMMENTED** [review]
This looks useful! It would be nice to upstream it so that other sequence to sequence models may make use of it. Also, you added it to `BartTokenizer` and not the fast tokenizer, is there a reason for that?

If we consider this to be a conversion to an s2s task, I think this would be better suited in a s2s processor like we have for `squad_convert_examples_to_features` or `glue_convert...`. I don't see any reason of having it linked to BART especially.

Pinging @thomwolf

**Comment by patil-suraj** [comment]
Hi @LysandreJik  
> and not the fast tokenizer, is there a reason for that?

No, just forgot to add that.

Upstream will be useful but we will need handle few cases differently for each seq2seq model i.e in case of t5 we manually need to add the deocder_start_token_id as T5 don't have a `bos` token. Also `eos' needs to be added manually. In case of mBart, it needs the language code as prefix token etc. And also AFAIK lot of people seem to be unfamiliar with the processors API

**Comment by patil-suraj** [comment]
hi @sshleifer , @LysandreJik any update ?

File: src/transformers/tokenization_bart.py
Comment (sshleifer) [inline src/transformers/tokenization_bart.py:88]: There is a new docstring on master/ tokenization_utils_base.py that you may want to (a) reuse or (b) modify.

**Comment by patil-suraj** [comment]
@sshleifer  updated the docs.

File: src/transformers/tokenization_bart.py
Comment (sgugger) [inline src/transformers/tokenization_bart.py:66]: The type annotations here were better before. The docstrings should not have abbreviations (and start with a capital and end with a full stop nit).

Comment (patil-suraj) [reply src/transformers/tokenization_bart.py:66]: Aah, I blindly copy pasted, will make the changes. Also can you tell me the where the doc error is coming from ?

Comment (sgugger) [reply src/transformers/tokenization_bart.py:66]: You're missing new lines before your lists I'd say.

**Review by LysandreJik — APPROVED** [review]
LGTM as-is (after the doc building test fixes), but we really should add the same method on the Fast tokenizer. Having parity on both tokenizers is one of our goals.

**Comment by patil-suraj** [comment]
@LysandreJik  I will add this for fast tokenizer too once this PR is merged.

**Comment by LysandreJik** [comment]
Sounds good!

**Comment by patil-suraj** [comment]
@LysandreJik , doc error is fixed, not sure if this failure is related to this PR.

**Pull request merged**
